### PR TITLE
Various expansions

### LINKED ISSUE #51: AWS ecosystem market map
More general overview of companies in AWS ecosystem.

Preferably a market map type graphic.

### PR DESCRIPTION
**Pull request by jlevy**
Variety of changes. See commits.

Fixes #51. Progress on #65.

@ThanosBaskous review?
@nitingithub review also welcome!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -105,6 +111,7 @@ General Information
 	-	and you can afford, manage, or negotiate its somewhat higher costs
 	-	... then AWS is likely a good option for your company.
 -	Each of those reasons above might point to situations where other services are preferable. In practice, many, if not most, tech startups as well as a number of modern large companies can or already do benefit from using AWS. Many large enterprises are partly migrating internal infrastructure to Azure, Google Cloud, and AWS.
+-	**Costs:** Billing and cost management are such big topics we have [an entire section on this](#billing-and-cost-management).
```
Comment (ThanosBaskous) [inline README.md]: Suggestion: "are such big topics that we have"

File: README.md [hunk 2]
```diff
@@ -127,8 +141,7 @@ General Information
...
 -	Most customers use a few services heavily, a few services lightly, and the rest not at all. What services you’ll use depends on your use cases. Choices differ substantially from company to company.
--	Just because AWS has a service that sounds promising, it doesn’t mean you should use it. Some services are very narrow in use case, not mature, are overly opinionated, or have limitations, so very few people use them. More on this next.
--	Many customers combine AWS with other non-AWS services. For example, legacy systems or secure data might be in a managed hosting provider, while other systems are AWS. Or a company might only use S3 with another provider doing everything else. However small startups or projects starting fresh will typically stick to AWS or Google Cloud only.
+-	**Immature and unpopular services:** Just because AWS has a service that sounds promising, it doesn’t mean you should use it. Some services are very narrow in use case, not mature, are overly opinionated, or have limitations, so building your own solution may be better. We try to give a sense for this by breaking products into categories.
```
Comment (ThanosBaskous) [inline README.md]: Also, some are semi-abandoned / in maintenance mode (e.g. SimpleDB). Probably not something for this ticket, but maybe we should find structured way of classifying AWS services in this regard (or at least call out it out explicitly). I also just noticed that we don't have SimpleDB section. I will add tickets for both issues.

File: README.md [hunk 3]
```diff
@@ -147,37 +160,37 @@ General Information
 	-	[EMR](#emr): Managed Hadoop
 	-	[Elasticsearch](https://aws.amazon.com/elasticsearch-service/): Managed Elasticsearch
 	-	[ElastiCache](https://aws.amazon.com/elasticache/): Managed Redis and Memcached
--	**Optional but important infrastructure:** These are key and useful infrastructure are less widely known used. You may have legitimate reasons to prefer alternatives, so evaluate with care you to be sure they fit your needs:
-	-	[Lambda](#lambda): Running small, fully managed tasks “serverless”
+-	**Optional but important infrastructure:** These are key and useful infrastructure that are less widely known used. You may have legitimate reasons to prefer alternatives, so evaluate with care you to be sure they fit your needs:
```
Comment (ThanosBaskous) [inline README.md]: Suggestion (2 changes in each): "This is key and useful infrastructure that is less widely known and used" or "These are key and useful infrastructure components that are less widely known and used".

File: README.md [hunk 4]
```diff
@@ -147,37 +160,37 @@ General Information
...
 	-	[CloudTrail](https://aws.amazon.com/cloudtrail/): AWS API logging and audit (often neglected but important)
-	-	🕍[CloudFormation](#cloudformation): Templatized configuration of collections of AWS resources
+	-	⛓🕍[CloudFormation](#cloudformation): Templatized configuration of collections of AWS resources
 	-	🕍[Elastic Beanstalk](https://aws.amazon.com/elasticbeanstalk/): Fully managed (PaaS) deployment of packaged Java, .NET, PHP, Node.js, Python, Ruby, Go, and Docker applications
-	-	🐥[EFS](https://aws.amazon.com/efs/): Network filesystem
-	-	🕍[ECS](#ecs): Docker container/cluster management. Note Docker can be used directly, without ECS.
-	-	[ECR](https://aws.amazon.com/ecr/): Hosted private Docker registry.
+	-	🐥⛓[EFS](https://aws.amazon.com/efs/): Network filesystem
+	-	⛓🕍[ECS](#ecs): Docker container/cluster management. Note Docker can be used directly, without ECS.
+	-	⛓[ECR](https://aws.amazon.com/ecr/): Hosted private Docker registry.
```
Comment (ThanosBaskous) [inline README.md]: Was already this way, but period at end is inconsistent with other list items.

File: README.md [hunk 5]
```diff
@@ -198,6 +211,16 @@ General Information
 	-	🕍[OpsWorks](https://aws.amazon.com/opsworks/): Management of your deployments using Chef. While Chef is popular, it seems few people use OpsWorks, since it involves going in on a whole different code deployment framework.
 -	[AWS in Plain English](https://www.expeditedssl.com/aws-in-plain-english) offers more friendly explanation of what all the other different services are.
 
+### Tools and Services Market Landscape
+
+There are now enough cloud and “big data” enterprise companies and products few can keep up with the market landscape. (See the [Big Data Evolving Landscape – 2016](https://practicalanalytics.co/2016/02/09/big-data-evolving-landscape-2016/) for one attempt at this.)
```
Comment (ThanosBaskous) [inline README.md]: Suggestion: "and products that few"

File: README.md [hunk 6]
```diff
@@ -507,7 +513,9 @@ We cover security basics first, since configuring user accounts is something you
 -	[**Trusted Advisor**](https://aws.amazon.com/blogs/aws/trusted-advisor-console-basic/) addresses a variety of best practices, but also offers some basic security checks around IAM usage, security group configurations, and MFA.
 -	**Use KMS for managing keys**: AWS offers [KMS](#kms) for securely managing encryption keys, which is usually a far better option than handling key security yourself. See [below](#kms).
 -	[**AWS WAF**](https://aws.amazon.com/waf) is a web application firewall to help you protect your applications from common attack patterns.
--	🔹**Export and audit security settings:** You can audit security policies simply by exporting settings using AWS APIs, e.g. using a Boto script like [SecConfig.py](https://gist.github.com/jlevy/cce1b44fc24f94599d0a4b3e613cc15d) (from [this 2013 talk](http://www.slideshare.net/AmazonWebServices/intrusion-detection-in-the-cloud-sec402-aws-reinvent-2013)) and then reviewing and monitoring changes manually or automatically.
+-	**Security auditing:**
+	-	[Security Monkey](https://github.com/Netflix/security_monkey) is an open source tool to assist with secuirty audits.
```
Comment (ThanosBaskous) [inline README.md]: Typo: "secuirty" -> "security"
Suggestion: "is an open source tool that is designed to assist with security audits."

File: README.md [hunk 7]
```diff
@@ -1217,6 +1206,51 @@ This section covers tips and information on achieving [high availability](https:
...
+-	Multiple Amazon accounts can be linked for billing purposes using the [Consolidated Billing](http://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/consolidated-billing.html) feature. Large enterprises may need complex billing structures depending on ownership and approval processes.
+
+### EC2 Cost Management
+
+-	With EC2, there is a trade-off between engineering effort (more analysis, more tools, more complex architectures) and spend rate on AWS. If your EC2 costs are small, many of the efforts here are not worth the engineering time required to make them work. But once you know your costs will be growing in excess of an engineer’s salary, serious investment is often worthwhile.
+-	🔹**Spot instances:**
+	-	EC2 [spot instances](https://aws.amazon.com/ec2/spot/) are a way to get EC2 resources at significant discount — often many times cheaper than standard on-demand prices — if you’re willing to accept the possibility that they be terminated with little to no warning.
+	-	Use spot instances for potentially very significant discounts whenever you can use resources that may be restarted and don’t maintain long-term state.
```
Comment (ThanosBaskous) [inline README.md]: Nit: I know you are just moving this (I may actually be the original culprit) but just noticed inconsistent capitalization ("spot" here, "Spot" below, "spot" below that).

Comment (jlevy) [inline README.md]: Looks like it is capitalized everywhere. Fixing.

**Comment by jlevy** [comment]
Think this addresses all the changes. Thanks for the close review.

**Pull request merged**